url8db16c7d62_Arsaell_EdT_tgz-pJ8-DATA_MainJ8/src/DATA/Filler.java

```diff
@@ -0,0 +1,980 @@
+						temp.put(i, c1 , con);
+						break;
+					}
+					
+					else if (i == temp.size() - 1)	{
+						//ps.println("addLast() : " + (temp.size() + 1));
+						temp.put(c1, con);
+						break;
+					}
+				}
+			}
+			
+			//On enlève l'élément inséré à l'initialisation.
+			if (hm == data.get(1))	{
+				temp.remove(data.get(data.size() - 1).getKey(data.get(data.size() - 1).size() - 1));
+			}
+		}
+
+		/*
+		//Boucle d'affichage uniquement, peut être passée sous commentaire
+		for (Constrainable c : res)	{
+			
+			double con = -1;
+			
+			for (HashMap<Constrainable, Double> hm : data)
+				if (hm.get(c) != null)	{
+					con = hm.get(c);
+					break;
+				}
+			
+			ps.println("Filler.orderByValues() #res : " + c + " \t " + con);
+		}
+		//*/
+		return temp;
+	}
+
+	/**
+	 * Les quatres méthodes suivantes sont simplement des interfaces entre la méthode fill() (appelante)
+	 * et la méthode takeCareOf(Group, Field) (qui fait la plus grande partie du travail)
+	 */
+	private boolean takeCareOf(Field f, ArrayList<Group> order)	{
+		
+		for (Group group : order)
+			if (group.getClasses().containsKey(f) && !group.getFieldsDone().get(f))
+				if (!this.takeCareOf(group, f))
+					return false;
+		return true;
+	}
+
+	private boolean takeCareOf(ClassType ct, ArrayList<Group> groupOrder, ArrayList<Field> fieldOrder)	{
+		
+		for (Group group : groupOrder)	{
+			for (Field field : fieldOrder)	{
+				if (field.getType() == ct && group.getClasses().containsKey(field) && !group.getFieldsDone().get(field))	{
+					if (!this.takeCareOf(group, field))
+						return false;
+				}
+			}
+		}
+		return true;
+	}
+	
+	private boolean takeCareOf(Teacher teach, ArrayList<Field> fieldsOrder, ArrayList<Group> groupsOrder)	{
+		
+		for (Field f : fieldsOrder)	{
+			if (teach.knows(f))	{
+				for (Group g : groupsOrder)	{
+					if (teach.getLinks().getLinks(g).size() != 0 && !(g.getFieldsDone().get(f)))	{
+
+						Time left = g.getClasses().get(f);
+						
+						for (Lesson l : g.getWeekTable().getSlotsConcerning(f)) 
+							left = left.substract(l.getDuration());
+
+						if (!this.takeCareOf(g, f))
+							return false;
+					}
+				}
+			}
+		}
+		return true;
+	}
+	
+	private boolean takeCareOf(Group g, ArrayList<Field> order)	{
+		
+		for (Field f : order)	{
+			if (g.classes.containsKey(f) && !g.getFieldsDone().get(f))	{
+				
+				if (!this.takeCareOf(g, f))
+					return false;
+			}
+		}
+		
+		return true;
+	}
+
+	/**
+	 * S'occupe de générer et attribuer des Lessons correspondant aux paramètres jusqu'à remplir les exigences de group.classes
+	 * Si cette fonction renvoie false, la méthode fill() qui l'appelle s'occupera de transmettre l'erreur à la méthode handleError().
+	 * @param group le Group dont la méthode attribue les matières
+	 * @param field la matière concernée
+	 * @return false si une erreur a été rencontrée (un descriptif de l'erreur est affiché dans la sortie terminal).
+	 */
+	private boolean takeCareOf(Group group, Field field) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 31. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test. We recommend that you simplify this method or break it into multiple methods.